docs/website/versioned_docs/version-3.0.0/development/devfile.md

```diff
@@ -0,0 +1,229 @@
+---
+title: Devfile Reference
+sidebar_position: 30
+---
+
+## Description
+
+### Overview
+
+We implement the [Devfile 2.2.0 reference](https://devfile.io/docs/devfile/2.2.0/user-guide/api-reference/) in `odo`.
+
+Below is `odo`-specific examples regarding the Devfile reference. Everything we have listed below is implemented in the current version of `odo`.
+
+This API Reference uses examples and snippets to help you create your own `devfile.yaml` file for your project.
+
+For this reference, we use one singular (working) Node.js example which can be ran on your own environment.
+
+### Why Devfile?
+
+### How `odo` Creates Resources from Devfile
+
+### File Synchronization
+
+### What Commands are Executed in Dev or Deploy
```

Comment:
Are you going to expand on these in this PR or in a later one?

Reply:
Yup! You just reviewed it too early (which is okay!) thank you for all the changes.